### PR TITLE
feat(demo): egjs demo Music Player 1.0.0 version

### PR DESCRIPTION
**Pull request by BillionaireDY**
feat(demo): egjs demo Music Player 1.0.0 version

### CODE REVIEW

File: demo/assets/js/musicplayer.js [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+	init() {
+        this.initPlayListPannel();
+		this.playMusic(0);
+        this.turnTableHandler();
+        this.listHandler();
+	}
+
+    // make list Dom to the bottom of music player
+	initPlayListPannel() {
+        let deg =  Math.round(360 / this.playList.length);
```
Comment (sculove) [inline demo/assets/js/musicplayer.js]: How about caching `this.playList.length` ?

File: demo/assets/js/musicplayer.js [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+        }).on({
+            "change": (e) => {
+                this.listCarouselDom.style[eg.Axes.TRANSFORM] = "translateZ("+ -1 * translateZ +"px) rotateY(" + e.pos.rotateX + "deg)";
+            },
+            "hold": (e) => {
+                this.pauseMusic();
+            },
+            "release": (e) => {
+                destMusicIndex = 0;
+                let currentRadian = e.destPos.rotateX * Math.PI / 180; // 현재라디안
```
Comment (sculove) [inline demo/assets/js/musicplayer.js]: delete korean comments

File: demo/assets/js/musicplayer.js [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+
+                // 자기자신의 범위라디안에 들어온 뮤직을 가운데로 위치
```
Comment (sculove) [inline demo/assets/js/musicplayer.js]: delete korean comments

File: demo/assets/js/musicplayer.js [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+                    circular: true
+                },
+                rotateY: {
+                    range: [0, 360],
+                    circular: true
+                }
+            },
+            deceleration: 0.0024
+        }).on({
+            "change": (e) => {
```
Comment (JaeYeopHan) [inline demo/assets/js/musicplayer.js]: how about remove this code? :)

File: demo/assets/js/musicplayer.js [hunk 5]
```diff
@@ -0,0 +1,254 @@
...
+                for(let i = 0; i < this.playList.length; i++) {
+                    if(unitRad * i - rangeRad <= currentRadian && currentRadian <= unitRad * i + rangeRad) {
+                        destMusicIndex = this.playList.length - i === this.playList.length ? 0 : this.playList.length - i;
+                        this.listCarouselDom.style[eg.Axes.TRANSFORM] = "translateZ("+ -1 * translateZ +"px) rotateY(" + uinitDeg * i + "deg)";
+                    }
+                }
+                this.playMusic(destMusicIndex);
+            },
+            "animationEnd": (e) => {
+                // 관성에 의해 더 간경우
```
Comment (sculove) [inline demo/assets/js/musicplayer.js]: delete korean comments or translate it

File: demo/assets/js/musicplayer.js [hunk 6]
```diff
@@ -0,0 +1,254 @@
...
+                this.listLocator(destMusicIndex);
+            }
+        }).mapInput(["rotateX", null], new eg.Axes.HammerInput("#list_carousel"))
+    }
+    
+    listLocator(locate) {
+        let uinitDeg =  360 / this.playList.length; // 단위각
```
Comment (sculove) [inline demo/assets/js/musicplayer.js]: delete korean comments or translate it

File: demo/assets/js/musicplayer.js [hunk 7]
```diff
@@ -0,0 +1,254 @@
...
+                this.pauseMusic();
+            },
+            "release": (e) => {
+                destMusicIndex = 0;
+                let currentRadian = e.destPos.rotateX * Math.PI / 180; // 현재라디안
+
+                // 자기자신의 범위라디안에 들어온 뮤직을 가운데로 위치
+                for(let i = 0; i < this.playList.length; i++) {
+                    if(unitRad * i - rangeRad <= currentRadian && currentRadian <= unitRad * i + rangeRad) {
+                        destMusicIndex = this.playList.length - i === this.playList.length ? 0 : this.playList.length - i;
```
Comment (sculove) [inline demo/assets/js/musicplayer.js]: you can use `%` operator like this.
```js
destMusicIndex = i % this.playList.length;
```

File: demo/assets/js/musicplayer.js [hunk 8]
```diff
@@ -0,0 +1,254 @@
...
+    
+    listLocator(locate) {
+        let uinitDeg =  360 / this.playList.length; // 단위각
+        let translateZ =  Math.round( ( 210  / 2 ) /  Math.tan( Math.PI / this.playList.length ));
+        let allocator = this.playList.length - locate;
+        this.listCarouselDom.style[eg.Axes.TRANSFORM] = "translateZ("+ -1 * translateZ +"px) rotateY(" + uinitDeg * allocator + "deg)";
+    }
+
+	turnTableHandler() {
+        let axes = new eg.Axes({
```
Comment (sculove) [inline demo/assets/js/musicplayer.js]: What is this? this code is meaningless.

Comment (sculove) [inline demo/assets/js/musicplayer.js]: If you use axes, you can use only one instance lie this.
```js
new eg.Axes({
            axis: {
                turnX: {
                    range: [0, 360],
                    circular: true
                },
                turnY: {
                    range: [0, 360],
                    circular: true
                },
                flickingX: {
                      // ...
                }
            },
```

File: demo/assets/html/musicplayer.html [hunk 9]
```diff
@@ -0,0 +1,31 @@
+<div class="musicplayer_wrapper">
+    <section class="musicplayer_appbar">
+        <img src="common/image/logo.svg" alt="egjs-logo" />
+        <img src="common/image/type_white.svg" alt="egjs-logo" />
+        <span> Music Player</span>
+    </section>
```
Comment (sculove) [inline demo/assets/html/musicplayer.html]: How about removing`musicplayer_appbar` section?

File: demo/assets/js/musicplayer.js [hunk 10]
```diff
@@ -0,0 +1,254 @@
...
+                rotateY: {
+                    range: [0, 360],
+                    circular: true
+                }
+            },
+            deceleration: 0.0024
+        }).on({
+            "change": (e) => {
+            },
+            "hold": (e) => {
```
Comment (JaeYeopHan) [inline demo/assets/js/musicplayer.js]: how about change to one line?
like this:
```js
"hold": () => this.pauseMusic(),
```

File: demo/assets/js/musicplayer.js [hunk 11]
```diff
@@ -0,0 +1,254 @@
...
+            },
+            deceleration: 0.0024
+        }).on({
+            "change": (e) => {
+                this.listCarouselDom.style[eg.Axes.TRANSFORM] = "translateZ("+ -1 * translateZ +"px) rotateY(" + e.pos.rotateX + "deg)";
+            },
+            "hold": (e) => {
+                this.pauseMusic();
+            },
+            "release": (e) => {
```
Comment (sculove) [inline demo/assets/js/musicplayer.js]: When released the carousel is just moved to destMusic.
How about moving carousel using animation?

File: demo/assets/js/musicplayer.js [hunk 12]
```diff
@@ -0,0 +1,254 @@
...
+                let currentRadian = e.destPos.rotateX * Math.PI / 180; // 현재라디안
+
+                // 자기자신의 범위라디안에 들어온 뮤직을 가운데로 위치
+                for(let i = 0; i < this.playList.length; i++) {
+                    if(unitRad * i - rangeRad <= currentRadian && currentRadian <= unitRad * i + rangeRad) {
+                        destMusicIndex = this.playList.length - i === this.playList.length ? 0 : this.playList.length - i;
+                        this.listCarouselDom.style[eg.Axes.TRANSFORM] = "translateZ("+ -1 * translateZ +"px) rotateY(" + uinitDeg * i + "deg)";
+                    }
+                }
+                this.playMusic(destMusicIndex);
```
Comment (sculove) [inline demo/assets/js/musicplayer.js]: When the animation is running, the music is playing.
it's not good.

File: demo/assets/js/musicplayer.js [hunk 13]
```diff
@@ -0,0 +1,254 @@
...
+            },
+            "hold": (e) => {
+                this.pauseMusic();
+            },
+            "release": (e) => {
+                destMusicIndex = 0;
+                let currentRadian = e.destPos.rotateX * Math.PI / 180; // 현재라디안
+
+                // 자기자신의 범위라디안에 들어온 뮤직을 가운데로 위치
+                for(let i = 0; i < this.playList.length; i++) {
```
Comment (sculove) [inline demo/assets/js/musicplayer.js]: you should check e.destPos.
if destPos is not equal depaPos you should pass this job to `animationEnd`.

**Comment by sculove** [comment]
Thanks @BillionaireDY I applied a part of your demo.

**Pull request closed without merging**